modules/gapi/src/streaming/onevpl/accelerators/accel_policy_cpu.cpp

```diff
@@ -0,0 +1,227 @@
+        // do not free key here: last surface will release it
+    }
+    pool_table.clear();
+    GAPI_LOG_INFO(nullptr, "destroyed");
+}
+
+void VPLCPUAccelerationPolicy::init(session_t session) {
+    GAPI_LOG_INFO(nullptr, "initialize session: " << session);
+}
+
+void VPLCPUAccelerationPolicy::deinit(session_t session) {
+    GAPI_LOG_INFO(nullptr, "deinitialize session: " << session);
+}
+
+VPLCPUAccelerationPolicy::pool_key_t
+VPLCPUAccelerationPolicy::create_surface_pool(size_t pool_size, size_t surface_size_bytes,
+                                              surface_ptr_ctr_t creator) {
+    GAPI_LOG_DEBUG(nullptr, "pool size: " << pool_size << ", surface size bytes: " << surface_size_bytes);
+
+    // create empty pool
+    pool_t pool;
+    pool.reserve(pool_size);
+
+    // allocate workplace memory area
+    size_t preallocated_raw_bytes = pool_size * surface_size_bytes;
+    size_t page_size_bytes = 4 * 1024;
+    void *preallocated_pool_memory_ptr = nullptr;
+
+#ifdef _WIN32
+    SYSTEM_INFO sysInfo;
+    GetSystemInfo(&sysInfo);
+    page_size_bytes = sysInfo.dwPageSize;
+
+    GAPI_LOG_DEBUG(nullptr, "page size: " << page_size_bytes << ", preallocated_raw_bytes: " << preallocated_raw_bytes);
+    preallocated_pool_memory_ptr = _aligned_malloc(preallocated_raw_bytes, page_size_bytes);
+#endif
+
+    if (!preallocated_pool_memory_ptr) {
+        throw std::runtime_error("VPLCPUAccelerationPolicy::create_surface_pool - failed: not enough memory."
+                                 "Requested surface count: " + std::to_string(pool_size) +
+                                 ", surface bytes: " + std::to_string(surface_size_bytes));
+    }
+
+    // fill pool with surfaces
+    std::shared_ptr<void> workspace_mem_owner (preallocated_pool_memory_ptr, [] (void *ptr){
+        GAPI_LOG_INFO(nullptr, "Free workspace memory: " << ptr);
+#ifdef _WIN32
+        _aligned_free(ptr);
+        GAPI_LOG_INFO(nullptr, "Released workspace memory: " << ptr);
+        ptr = nullptr;
+#else
+        abort(); //not implemented
+#endif
+
+        });
+    size_t i = 0;
+    try {
+        for (; i < pool_size; i++) {
+            size_t preallocated_mem_offset = static_cast<size_t>(i) * surface_size_bytes;
+
+            surface_ptr_t surf_ptr = creator(workspace_mem_owner,
+                                             preallocated_mem_offset,
+                                             preallocated_raw_bytes);
+            pool.push_back(std::move(surf_ptr));
+        }
+    } catch (const std::exception& ex) {
+        throw std::runtime_error(std::string("VPLCPUAccelerationPolicy::create_surface_pool - ") +
+                                 "cannot construct surface index: " + std::to_string(i) + ", error: " +
+                                 ex.what() +
+                                 "Requested surface count: " + std::to_string(pool_size) +
+                                 ", surface bytes: " + std::to_string(surface_size_bytes));
+    }
```

Comment:
Do we really need this handling here? What throws here in particular - is it our code or the external one?

Any exception handling within our module complicates the future debug (or I miss something)

Reply:
ok let's see in the future PRs